### PR TITLE
Edge detection using open cv

### LINKED ISSUE #589: Edge Detction of a image using OpenCV
# Aim

In this Script, we going to detect the edges of an image using a python script.

# Details

In our script, the user has to just upload a picture and as an output, he/she will get an edge-detected picture. For this purpose, we are going to use OpenCV.
It is very useful because edges are the points in images where the color or brightness has great changes.

### PR DESCRIPTION
**Pull request by ShubhamGupta577**
# Description

Adding a script uses OpenCV for taking input and output for the image. In this, we are using the Canny algorithm for the detection of the edge

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by ShubhamGupta577** [comment]
@kaustubhgupta Please Review this.

File: Edge Detection/Readme.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+### You can use this in two ways
+* If your machine has any python intrepretor than you can go for [**Edge_Detection.py**](https://github.com/ShubhamGupta577/Amazing-Python-Scripts/blob/Edge-Detection-Using-OpenCV/Edge%20Detection/Edge_Detection.py)
+  * For these you need to install **OpenCV** in your machine.
+* Another way is to use the [**Google Colab file**](https://github.com/ShubhamGupta577/Amazing-Python-Scripts/blob/Edge-Detection-Using-OpenCV/Edge%20Detection/Edge_Detection.ipynb) which will be excecute on a cloud.
+
+## Technologies used
+<img src="https://img.shields.io/badge/Python-14354C?style=for-the-badge&logo=python&logoColor=white"/> <img src="https://img.shields.io/badge/OpenCV-0099E5?style=for-the-badge&logo=opencv&logoColor=white"/> 
+<img src="https://img.shields.io/badge/Google Colab-F9AB00?style=for-the-badge&logo=Google-Colab&logoColor=white"/> 
+
+## Sample Input/Output
```
Comment (kaustubhgupta) [inline Edge Detection/Readme.md:15]: Images not rendered

File: Edge Detection/Edge_Detection.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#Printing the original image
+cv.imshow('Original',img)
+#Reducing the noise from the image
+gray=cv.cvtColor(img,cv.COLOR_RGB2GRAY)
+
+#Using Canny algorithm to detect the edges of the image
+final=cv.Canny(gray,100,200)
+print('Number of edges'+ '=' +str(len(final)))     #printing Number of edges
+cv.imshow('Final',final)
+cv.waitKey(0)
```
Comment (kaustubhgupta) [inline Edge Detection/Edge_Detection.py:21]: Add this line at the end of the code: cv.destroyAllWindows()

File: Edge Detection/Edge_Detection.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'''
+  For using this Script you need to install OpenCV in your machine
+ '''
+#Importing openCV library
+import cv2 as cv
+
+#Taking path of input from the user
+path=input("Enter the path of uploaded image")
```
Comment (kaustubhgupta) [inline Edge Detection/Edge_Detection.py:8]:
```suggestion
path=input("Enter the path of uploaded image: ")
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by ShubhamGupta577** [comment]
Requested changes have been made.

File: Edge Detection/Readme.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+* If your machine has any python intrepretor than you can go for [**Edge_Detection.py**](https://github.com/ShubhamGupta577/Amazing-Python-Scripts/blob/Edge-Detection-Using-OpenCV/Edge%20Detection/Edge_Detection.py)
+  * For these you need to install **OpenCV** in your machine.
+* Another way is to use the [**Google Colab file**](https://github.com/ShubhamGupta577/Amazing-Python-Scripts/blob/Edge-Detection-Using-OpenCV/Edge%20Detection/Edge_Detection.ipynb) which will be excecute on a cloud.
+
+## Technologies used
+* **Python**
+* **OpenCV**
+* **Google Colab**
+## Sample Input/Output
+<img src ="https://github.com/ShubhamGupta577/Testing/blob/main/Sample_Input.jpg" width=400/> <img src ="https://github.com/ShubhamGupta577/Testing/blob/main/Sample_Output.png" width=400/>
```
Comment (kaustubhgupta) [inline Edge Detection/Readme.md:16]: Images still not rendered 
image

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
@ShubhamGupta577 Have a look at suggested changes if you haven't seen them

File: Edge Detection/Readme.md (hunk 4 above)
Comment (antrikshmisri) [reply Edge Detection/Readme.md:16]: The image links are broken

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

**Comment by ShubhamGupta577** [comment]
@kaustubhgupta , @antrikshmisri , Requested changes had done. Sorry for the delay. Please have a look into it.

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**